backend/lib/edgehog/containers/image_credentials/base64_json_encoding.ex

```diff
@@ -0,0 +1,38 @@
+#
+# This file is part of Edgehog.
+#
+# Copyright 2024 SECO Mind Srl
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+defmodule Edgehog.Containers.ImageCredentials.Base64JsonEncoding do
+  @moduledoc false
+  use Ash.Resource.Calculation
+
+  alias AshGraphql.Types.JSON
+
+  def calculate(records, _opts, _context) do
+    Enum.map(records, fn record ->
+      with {:ok, username} <- Map.fetch(record, :username),
+           {:ok, password} <- Map.fetch(record, :password) do
+        data = [username: username, password: password]
+        json = JSON.encode(data)
+
+        Base.encode64(json)
+      end
+    end)
```

Comment:
Just like in #645, we're dealing with structs so we already know their attributes.
nit/tip: pipelines are your friend
nit/tip 2: sometimes the code is (arguably) clearer using for comprehensions

```suggestion
    for record <- records do
      %{username: record.username, password: record.password}
      |>  JSON.encode()
      |> Base.encode64()
    end
```